pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -444,6 +444,15 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 							}
 							perRouteFilters[util.StatefulSessionFilter] = protoconv.MessageToAny(perRouteStatefulSession)
 						}
+					} else if node.Metadata.EnableHTTPHeaderSession {
```

Comment:
Is this intended?  service label takes priority over node meta

Reply:
Yes. Because service has specified annotation - we want to honour that cookie.